exporter/kafkaexporter/factory_test.go

```diff
@@ -122,15 +122,81 @@ func TestWithMarshalers(t *testing.T) {
 	})
 }
 
-type customMarshaler struct {
+type customTracesMarshaler struct {}
+
+var _ TracesMarshaler = (*customTracesMarshaler)(nil)
+
+func (c customTracesMarshaler) Marshal(_ pdata.Traces, topic string) ([]*sarama.ProducerMessage, error) {
+	panic("implement me")
+}
+
+func (c customTracesMarshaler) Encoding() string {
+	return "custom"
+}
+
+func TestWithLogsMarshalers(t *testing.T) {
+	cm := &customLogsMarshaler{}
+	f := NewFactory(WithLogsMarshalers(cm))
+	cfg := createDefaultConfig().(*Config)
+	// disable contacting broker
+	cfg.Metadata.Full = false
+
+	t.Run("custom_encoding", func(t *testing.T) {
+		cfg.Encoding = cm.Encoding()
```

Comment:
I think this tests actually are only somewhat useful. They only check that `Encoding()` function yields a result and factory accepts WithLogsMarshalers/WithMetricMarshalers.

TBH, the currently existing test (the only one which calls WithTracesMarshallers) is not very useful either for the same reasons. 

Perhaps that's OK (what's your guidance on this @MovieStoreGuy @pavolloffay?). I was thinking that maybe some sorts of tests matching non default marshaler encoding could be created instead, but perhaps it's too much of a stretch for a relatively simple API change here

Reply:
Apologies, didn't mean to leave this for so long. Github notifications are not my friend for some time now.

Currently late evening atm, so I come to review back tomorrow to full understand what is happening.

Reply:
Yes, I agree with @pmm-sumo.

The minimal required testing of custom encodings is that the encoded data matches a known output.
Testing that the set encoding matches the expected result is just the start.